docs/public/components/creating-components.md

```diff
@@ -58,27 +52,32 @@ This file hierarchy aimed at solving a few problems:
 * A main that imports a component should be able to select a specific implementation without compiling with the others.
   For example: the ZSTD library should not be included at compile time when the ZIP version is used.
 
-!!! warning "Important"
-    For all these reasons, components should never be nested.
 
-## Defining Components
+**Go package naming convention**:
 
-You can use the [invoke](../setup.md#invoke) task `inv components.new-component comp/<bundleName>/<component>` to generate a scaffold for your new component.
+All implementations must use the package name: `<component name>impl`.
+
+For example, a compression component with 2 implementation would use `package compressionimpl` in both
+`comp/<bundle>/compression/impl-zstd` and `comp/<bundle>/compression/impl-zip` folders.
+
```

Comment:
Actually, after trying to migrate a package, this is so annoying as it force import aliasing everywhere and we only import implementation `fx` folder and main. We should remove this